ci/forgive-me.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -x
+
+here="$(realpath "$(dirname "$0")")"
+source_dir="${here:?}/.."
+
+committish="${1:?}"
+tarballBaseName="${2:-solana-release}"
+remote=https://github.com/solana-labs/solana.git
+
+cleanup_tmp_source_dir() {
+  rm -rf "$tmp_source_dir"
+}
+
+set -e
+tmp_source_dir="$(mktemp --directory --tmpdir="$source_dir")"
+
+trap 'cleanup_tmp_source_dir' EXIT
+
+# Clone the specified committish
+git clone "$remote" "$tmp_source_dir"
+
+if pushd "$tmp_source_dir"; then
+  (
+    set +e
+
+    git fetch origin "$committish":branch_to_build
+    git checkout branch_to_build
```

Comment:
this can be merged into https://github.com/solana-labs/solana/pull/12337/files#r493804567